File: 0228-summary-ranges/0228-summary-ranges.java
class Solution {
    public List<String> summaryRanges(int[] nums) {
        List<String> result = new ArrayList<>();
        int lengthNums = nums.length;
        if(lengthNums == 0) return result;
        if(lengthNums == 1) {
            result.add(nums[0]+"");
            return result;
        }
        int left = nums[0]-1;
        int right = nums[0];

        for (int i = 1; i <= lengthNums; i++){
            if(nums[i-1] == right+1){
                right = nums[i-1];
                continue;
            }
            if(left == right)   result.add(left + "");
            else                result.add(left + "->" + right);
            
            left =  nums[i-1];
            right =  nums[i-1];

        }
        


        return result;
        
    }
}

File: 0228-summary-ranges/0228-summary-ranges.js
/**
 * @param {number[]} nums
 * @return {string[]}
 */
var summaryRanges = function(nums) {

    const result = [];
    const lengthNums = nums.length;
    if (lengthNums === 0) return result;

    let left = nums[0];

    for (let i = 1; i < lengthNums; i++) {
        // If the current number is not consecutive, add the range to the result
        if (nums[i] !== nums[i - 1] + 1) {
            result.push(left === nums[i - 1] ? 
                        `${left}` : 
                        `${left}->${nums[i - 1]}`);
            left = nums[i]; // Update the start of the next range
        }
    }

    // Add the last range
    result.push(left === nums[lengthNums - 1] ? 
                `${left}` : 
                `${left}->${nums[lengthNums - 1]}`);

    return result;
    
};